src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -110,18 +110,11 @@ public static class TarUtils
             Action actionOnFailure,
             CancellationToken cancellationToken)
         {
-            var processTcs = new TaskCompletionSource<int>();
-            using (var cancelSource = new CancellationTokenSource())
-            using (var linkedSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, cancelSource.Token))
+            using (var linkedSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
```

Comment:
This is not needed anymore

Reply:
Don't we need this for the other way round, if the download fails, we need to cancel the tarring process as well?

Reply:
Since we never call linkedSource.Cancel(), linkedSource is just a no-op wrapper

Reply:
👍